core/include/sched.h

```diff
@@ -81,15 +81,60 @@
 #define SCHED_H
 
 #include <stddef.h>
+#include <inttypes.h>
+
 #include "kernel_defines.h"
-#include "kernel_types.h"
 #include "native_sched.h"
 #include "clist.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @def MAXTHREADS
+ * @brief The maximum number of threads to be scheduled
+ */
+#ifndef MAXTHREADS
+#define MAXTHREADS 32
+#endif
+
+/**
+ * Canonical identifier for an invalid PID.
+ */
+#define KERNEL_PID_UNDEF 0
+
+/**
+ * The first valid PID (inclusive).
+ */
+#define KERNEL_PID_FIRST (KERNEL_PID_UNDEF + 1)
+
+/**
+ * The last valid PID (inclusive).
+ */
+#define KERNEL_PID_LAST (KERNEL_PID_FIRST + MAXTHREADS - 1)
+
+/**
+ * Macro for printing formatter
+ */
+#define PRIkernel_pid PRIi16
+
+/**
+ * Unique process identifier
+ */
+typedef int16_t kernel_pid_t;
```

Comment:
What meaning do negative pids have?

Reply:
I think there's a KERNEL_PID_UNDEF that's -1? Anyhow, this is all copied and otherwise unchanged.

Reply:
You just moved [`KERNEL_PID_UNDEF`](https://github.com/RIOT-OS/RIOT/pull/15481/files#diff-2ca01a325111fe84adb41a551530813506b0e34487598e4e75ce640573ca7c35R103) :wink: 

It's 0 - so that's why I'm wondering what negative PIDs mean.